### PR TITLE
Definition for inflected forms

### PR DESCRIPTION
**Pull request by Madoshakalaka**
## Major Change

- In the importing process, definitions for inflected forms are now correctly attributed to individual inflected wordforms, instead of lemmas.

## Minor Change

- filed `lemma` of model `Definition` now have `related_name=definitions`. `wordform.definitions` should be used now instead of `wordform.definition_set`

- Added unit test for lemma merging in importing process. Added integration test to make sure lemma nipaw only show 2 correct definition.

## Note

Before merging this I need to

- change `push-db` script to fix permission issues
- build and push new database to Sapir

### CODE REVIEW

**Comment by Madoshakalaka** [comment]
- and I need to fix ci real quick

**Comment by aarppe** [comment]
As far as I understand, the overall design has rather been that the dictionaries and their electronic versions contain Cree dictionary entries (whatever they are), for which English translations are given. Sometimes, these dictionary entries are lemmas, sometimes they are inflected forms, and sometimes they are multiword expressions. If the dictionary entry is a lemma, then POS information and paradigms should be provided.  This isn't necessarily the case for inflected forms not for multiword expressions.

On the longer term, dictionary entries that are inflected forms can be linked with the dictionary entry that represents the lemma given by the linguistic analysis of the inflected form.

But the reason for this approach is that by default one does not have an FST for analysis nor generation, and often the first FST is only partially implemented, at least in the initial stages. Thus, for a new language, one may not be able to rely on the FST to link dictionary entries that are inflected forms with ones that are lemmas.

**Comment by eddieantonio** [comment]
@Madoshakalaka please fix the test failure before review

File: CreeDictionary/API/admin.py [hunk 1]
```diff
@@ -50,6 +48,8 @@ def queryset(self, request, queryset):
             return queryset.filter(as_is=True)
 
 
+# todo: make admin search "extended"
+# todo: show inflection definition instead of lemma definition
```
Comment (Madoshakalaka) [inline CreeDictionary/API/admin.py]: about the first todo. I searched for "nipa" on the admin console when I was debugging. the result "nipâ" didn't show up until I added the circumflex. Considering it's not worth a full issue and not important, I'll just add a todo here.

**Comment by Madoshakalaka** [comment]
@aarppe Thanks for the insight, I'll keep these philosophy in mind.

File: CreeDictionary/DatabaseManager/xml_importer.py [hunk 2]
```diff
@@ -334,50 +337,19 @@ def import_xmls(dir_name: Path, multi_processing: int = 1, verbose=True):
 
     # these two will be imported to the database
     as_is_xml_lemma_pos_lc = []  # type: List[Tuple[str, str, str]]
-    true_lemma_analyses_to_xml_lemma_pos_lc = (
-        dict()
+    true_lemma_analyses_to_xml_lemma_pos_lc = defaultdict(
+        list
     )  # type: Dict[str, List[Tuple[str, str, str]]]
```
[the remark this answers is not in the document]
Comment (Madoshakalaka) [reply CreeDictionary/DatabaseManager/xml_importer.py]: yes that's why I wrote "todo: refactor and use xml tuple"

**Pull request merged**